service/history/api/startworkflow/api.go

```diff
@@ -143,45 +151,35 @@ func (s *Starter) prepare(ctx context.Context) error {
 		&request.WorkflowIdReusePolicy,
 		&request.WorkflowIdConflictPolicy)
 
-	api.OverrideStartWorkflowExecutionRequest(request, metrics.HistoryStartWorkflowExecutionScope, s.shardContext, metricsHandler)
+	api.OverrideStartWorkflowExecutionRequest(
+		request,
+		metrics.HistoryStartWorkflowExecutionScope,
+		s.shardContext,
+		s.shardContext.GetMetricsHandler(),
+	)
 
 	err := api.ValidateStartWorkflowExecutionRequest(ctx, request, s.shardContext, s.namespace, "StartWorkflowExecution")
 	if err != nil {
 		return err
 	}
 
 	if request.RequestEagerExecution {
-		metrics.WorkflowEagerExecutionCounter.With(
-			workflow.GetPerTaskQueueFamilyScope(
-				metricsHandler, s.namespace.Name(), request.TaskQueue.Name, s.shardContext.GetConfig(),
-				metrics.WorkflowTypeTag(request.WorkflowType.Name),
-			),
-		).Record(1)
+		metrics.WorkflowEagerExecutionCounter.With(s.metricsHandler).Record(1)
 
 		// Override to false to avoid having to look up the dynamic config throughout the different code paths.
 		if !s.shardContext.GetConfig().EnableEagerWorkflowStart(s.namespace.Name().String()) {
-			s.recordEagerDenied(eagerStartDeniedReasonDynamicConfigDisabled)
+			metrics.WorkflowEagerExecutionDeniedCounter.With(s.metricsHandler).
+				Record(1, metrics.ReasonTag(eagerStartDeniedReasonDynamicConfigDisabled))
 			request.RequestEagerExecution = false
-		}
-		if s.request.FirstWorkflowTaskBackoff != nil && s.request.FirstWorkflowTaskBackoff.AsDuration() > 0 {
```

Comment:
This could be double counting eager execution denied, so I changed to `else if`.

Reply:
what about the nil check? not needed anymore?

Reply:
Not needed, if it's nil, it will return 0.